src/switch_snmp.py

```diff
@@ -0,0 +1,234 @@
+
+    if switch_model.find('ExtremeXOS') != -1:
+        port_name_dict = get_snmp_table(switch, community,
+                                        port_name_oid, 1000000)
+        switch_type = 'Extreme'
+        trunk_group_tab = None
+        skip = 1000000
+
+    port_num = 0
+    snmp_tables = {}
+    for counter, snmp_oid in COUNTERS.items():
+        snmp_tables[counter] = get_snmp_table(
+            switch, community, snmp_oid, skip
+        )
+
+    # Let's consider that "now" is after all values are
+    # read for this particular switch.
+    now = str(int(time.time()))
+
+    for port_index, port_name in port_name_dict.iteritems():
+        # Skip bad ports. This can be a VLAN on Cisco device.
+        if (
+            port_index not in snmp_tables['bytesIn']
+            or
+            port_index not in snmp_tables['pktsIn']
+        ):
+            continue
+
+        counters = {}
+        for counter in COUNTERS:
+            counters[counter] = int(snmp_tables[counter][port_index])
+
+        supported_port = False
+        # Is it a Trunk / etherchannel?
+        if (
+            (switch_type == 'Dell' and re.match('^Po[0-9]+$', port_name))
+            or
+            (switch_type == 'ProCurve' and re.match('^Trk[0-9]+$', port_name))
+        ):
+            # On Dell devices all portchannels are listed, so skip those unused
+            if switch_type == 'Dell' and port_index not in active_trunks:
+                continue
+
+            supported_port = True
+
+        # Is it a real port?
+        elif (
+            (
+                switch_type == 'Dell'
+                and
+                re.match('^(Gi|Te)[0-9]/[0-9]/[0-9]+$', port_name)
+            )
+            or
+            (switch_type == 'ProCurve' and re.match('^[0-9]+$', port_name))
+            or
+            (switch_type == "Extreme")
+        ):
+
+            # Get rid of unsupported characters from port name
+            port_name = port_name.replace('/', '_').replace(':', '_')
+
+            if switch_type == 'ProCurve':
+                port_num = int(port_name)
+            elif switch_type == 'Dell':
+                # On Dell blade switches ports have non-contignous numbers,
+                # there is space for extra 4 ports after each 20.
+                port_num += 1
+
+            supported_port = True
+
+        else:
+            supported_port = False  # dont work on unknown switches
+
+        # This means that this is a supported type of port and we want to
+        # send its counters to Graphite
+        if supported_port:
+            for counter in COUNTERS:
+                graphite += "switches.{}.{}-{}.{} {} {}\n".format(
+                    switch, port_index, port_name,
+                    counter, counters[counter], now
+                )
+
+    print(graphite)
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument(
+        '-c', '--community', type=str, required=True,
```

Comment:
I don't like `--` arguments being "required".  I think `'public'` would be a good default for this.